logger/level.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package logger
+
+import "fmt"
+
+type Level uint8
+
+func (level Level) String() string {
+	switch level {
+	case Trace:
+		return "trace"
+	case Debug:
+		return "debug"
+	case Info:
+		return "info"
+	case Warn:
+		return "warn"
+	case Error:
+		return "error"
+	case Fatal:
+		return "fatal"
+	case Panic:
+		return "panic"
+	case Off:
+		return "off"
+	}
+
+	return "unknown"
+}
+
+// Convert a string to a Level, the reverse of Level.String
+func Parse(lvl string) (Level, error) {
+	switch lvl {
+	case "off":
+		return Off, nil
+	case "panic":
+		return Panic, nil
+	case "fatal":
+		return Fatal, nil
+	case "error":
+		return Error, nil
+	case "warn":
+		return Warn, nil
+	case "info":
+		return Info, nil
+	case "debug":
+		return Debug, nil
+	case "trace":
+		return Trace, nil
+	}
+
+	var l Level
+	return l, fmt.Errorf("not a valid Level: %q", lvl)
+}
+
+const (
```

Comment:
Initially it was because I wanted the Level to default to Debug when unset, but I don't think the level direction matters. The levels should be refereed to by name/const.

`unset` is not exported because it has no meaning to the outside world and it's an error to call `factory.SetLevel("name", unset)`.